spec.bs

```diff
@@ -1545,7 +1545,33 @@ For now, see
 <div algorithm>
   The <dfn for=Observable method><code>finally(|callback|)</code></dfn> method steps are:
 
-    1. <span class=XXX>TODO: Spec this and use |callback|.</span>
+    1. Let |sourceObservable| be [=this=].
+
+    1. Let |observable| be a [=new=] {{Observable}} whose [=Observable/subscribe callback=] is an
+       algorithm that takes a {{Subscriber}} |subscriber| and does the following:
+
+       1. Run |subscriber|'s {{Subscriber/addTeardown()}} method with |callback|.
+
+       1. Let |sourceObserver| be a new [=internal observer=], initialized as follows:
```

Comment:
You can just pass the `subscriber` directly to the `subscribe`. There's no need to create another reference in this one-off case.

```ts
Observable.prototype.finally = (callback) => {
  return new Observable(subscriber => {
    subscriber.addTeardown(callback)
    this.subscribe(subscriber, { signal: subscriber.signal })
  })
}
```

Reply:
Oops, I forgot you're in a differently typed environment. I retract my comment. It's fine.